File: CS/EmployeeAsUserExample.Module/BusinessObjects/EmployeeRole.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp.Security.Strategy;
using DevExpress.Xpo;
using DevExpress.Persistent.Base;

using DevExpress.Persistent.BaseImpl.PermissionPolicy;

namespace EmployeeAsUserExample.Module.BusinessObjects {
    [ImageName("BO_Role")]
    public class EmployeeRole : PermissionPolicyRoleBase, IPermissionPolicyRoleWithUsers {
        public EmployeeRole(Session session)
            : base(session) {
        }
        [Association("Employees-EmployeeRoles")]
        public XPCollection<Employee> Employees {
            get {
                return GetCollection<Employee>("Employees");
            }
        }
        IEnumerable<IPermissionPolicyUser> IPermissionPolicyRoleWithUsers.Users {
            get { return Employees.OfType<IPermissionPolicyUser>(); }
        } 
    }
}

File: VB/EmployeeAsUserExample.Module/BusinessObjects/EmployeeRole.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp.Security.Strategy
Imports DevExpress.Xpo
Imports DevExpress.Persistent.Base

Imports DevExpress.Persistent.BaseImpl.PermissionPolicy

Namespace EmployeeAsUserExample.Module.BusinessObjects
    <ImageName("BO_Role")> _
    Public Class EmployeeRole
        Inherits PermissionPolicyRoleBase
        Implements IPermissionPolicyRoleWithUsers

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub
        <Association("Employees-EmployeeRoles")> _
        Public ReadOnly Property Employees() As XPCollection(Of Employee)
            Get
                Return GetCollection(Of Employee)("Employees")
            End Get
        End Property
        Private ReadOnly Property IPermissionPolicyRoleWithUsers_Users() As IEnumerable(Of IPermissionPolicyUser) Implements IPermissionPolicyRoleWithUsers.Users
            Get
                Return Employees.OfType(Of IPermissionPolicyUser)()
            End Get
        End Property
    End Class
End Namespace
